hknweb/forms.py

```diff
@@ -236,30 +236,26 @@ def generate_password() -> str:
 
             return password
 
-        email_information = []
+        # Process user data into useable structs
+        user_data = []
         for row in rows:
-            # If username is None or already exists, skip provisioning
-            if (row["username"] is None) or (row["username"] in existing_usernames):
+            if row["username"] is None or row["username"] in existing_usernames:
                 continue
-
-            # Generate a password
-            password = generate_password()
-
-            # Construct user object
-            user = User.objects.create_user(
+            user_data.append(User(
                 username=row["username"],
                 first_name=row["First name"],
                 last_name=row["Last name"],
                 email=row["Berkeley email"],
-                password=password,
-            )
-            user.save()
-
-            # Add user to the candidates group
+                password=generate_password(),
```

Comment:
Password needs to be hashed. `create_user` handles this, but without it we need to handle it ourselves

Reply:
Probably relevant: https://docs.djangoproject.com/en/4.2/topics/auth/passwords/#django.contrib.auth.hashers.make_password

Reply:
Sounds good; is it a salted hash by any chance or just a standard hash?

Reply:
On second look, there's also https://docs.djangoproject.com/en/4.2/ref/contrib/auth/#django.contrib.auth.models.User.set_password which might be better here

Reply:
(with make_password I think passing in None for the salt makes it use the default option which is probably fine, not sure what we're running in production)

Reply:
Sounds good.  I'm guessing that the password hashing algorithm uses an intentionally slow algorithm so it slows things down a bit now, but the process is still a bit faster than before.

Reply:
Alright, should be good now.  Curious if bulk_create() also hashes the password for you (which would make set_password() unnecessary), but probably not since the overhead with the PBKDF alg slowing things down is just always gonna be there unfortunately.